velox/functions/prestosql/StringFunctions.h

```diff
@@ -96,6 +96,17 @@ struct Md5Function {
   }
 };
 
+/// sha1(varbinary) -> varbinary
+template <typename T>
```

Comment:
Since there is only one signature using Varbinary type, the template variables should not be needed. But I notice other similar functions also using template variables. Please can you check if the code compiles without the use of template variables.

Reply:
I have checked and the code fails to compile without this template declaration. 
- If I try to remove the template declaration: `error: no member named 'udf' in 'facebook::velox::functions::Sha1Function'`
- If I try to remove the template altogether: `error: no member named 'resolver' in 'facebook::velox::Varbinary'`
- If I try to set the template to Varbinary directly: `error: use of undeclared identifier 'T'`

I also double-checked the doc and it states that all functions must be templated:
> All simple function classes need to be templated, and provide a “call” method (or one of the variations described below). The top-level template parameter provides the type system adapter, which allows developers to use non-primitive types such as strings, arrays, maps, and struct (check below for examples). Although the top-level template parameter is not used for functions operating on primitive types, such as the one in the example above, it still needs to be specified.

Reply:
There seems like a misunderstanding. Should've clarified. I didn't mean the template at the topmost level. That is indeed needed.

I was more curious about the template usage in the parameters of the call function. TTo and TFrom needn't be used here I think.

template <typename TTo, typename TFrom>
  FOLLY_ALWAYS_INLINE void call(TTo& result, const TFrom& input) {
    stringImpl::sha1(result, input);
  }

Did you check these ?

Reply:
Oh, yes. We can, in fact we did it for spooky_hash functions, e.g. 
```c++
FOLLY_ALWAYS_INLINE void call(out_type<Varbinary>& result, const arg_type<Varbinary>& input) {
  ...
}
```

Would you like me to update? I can update this function and open a follow-up to fix all of the other ones.

Reply:
Yeah, lets update this function.